src/compiler/binder.ts

```diff
@@ -1751,4 +1791,437 @@ namespace ts {
             implicitLabels = [];
         }
     }
+
+    /**
+     * Computes the transform flags for a node, given the transform flags of its subtree
+     *
+     * @param node The node to analyze
+     * @param subtreeFlags Transform flags computed for this node's subtree
+     */
+    export function computeTransformFlagsForNode(node: Node, subtreeFlags: TransformFlags) {
+        // Ambient nodes are TypeScript syntax and the flags of their subtree are ignored.
+        if (node.flags & NodeFlags.Ambient) {
+            return (node.transformFlags = TransformFlags.AssertTypeScript)
+                & ~(node.excludeTransformFlags = TransformFlags.NodeExcludes);
+        }
+
+        // Mark transformations needed for each node
+        let transformFlags: TransformFlags;
+        let excludeFlags: TransformFlags;
+        switch (node.kind) {
+            case SyntaxKind.PublicKeyword:
+            case SyntaxKind.PrivateKeyword:
+            case SyntaxKind.ProtectedKeyword:
+            case SyntaxKind.AbstractKeyword:
+            case SyntaxKind.DeclareKeyword:
+            case SyntaxKind.AsyncKeyword:
+            case SyntaxKind.ConstKeyword:

```

Comment:
The only ES6 `const` is for `VariableDeclarationList`, which is not stored as a modifier. The only time `const` is a `Modifier` is for a const enum, which is TypeScript only.